packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -343,6 +343,34 @@ export function moveCollection(
   return applyEdit(edit);
 }
 
+export function renameCollection(
+  fromNS: string,
+  toNS: string
+): DataModelingThunkAction<
+  void,
+  ApplyEditAction | ApplyEditFailedAction | CollectionSelectedAction
+> {
+  return (dispatch) => {
+    const edit: Omit<
+      Extract<Edit, { type: 'RenameCollection' }>,
+      'id' | 'timestamp'
+    > = {
+      type: 'RenameCollection',
+      fromNS,
+      toNS,
+    };
+
+    dispatch(applyEdit(edit));
+
+    // When renaming we want to keep the renamed collection selected in the UI,
+    // the ID is the collection name.
+    dispatch({
+      type: DiagramActionTypes.COLLECTION_SELECTED,
+      namespace: toNS,
+    });
```

Comment:
Batch updates are an antipattern, reducer should be the one accounting for that instead of multiple events being dispatched